### ISSUE TITLE
[Elixir] Implement concept exercise: nil-values

### ISSUE THREAD

**Issue by neenjaw** [exercism/v3#1487]
This issue describes how to implement the `nil-values` concept exercise for the Elixir track.

[🔗 **How to help**](#actionable-items)

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

This is a fork from the [C# nullability exercise](https://github.com/exercism/v3/blob/master/languages/csharp/exercises/concept/nullability/.docs/instructions.md) which uses the [Name Badge Story](https://github.com/exercism/v3/blob/master/reference/stories/nullability.name-badges.md)

The goal of this exercise is to teach how to use `nil` values for when nothing exists as a value

## Learning objectives

- Know what `nil` represents
- Know that `nil` is a special atom
- Know how to pattern match on `nil`

## Out of scope

- Unsure

## Concepts

- `nil-values`

## Prerequisites

- `booleans`
- `strings`
- `pattern-matching`

## Narrative for exercise

Following the [Name Badge Story](https://github.com/exercism/v3/blob/master/reference/stories/nullability.name-badges.md), students should be tasked to create a _name badge_ substituting `nil` values for default options.

The task should be broken into sub-components which then culminate in the lesson.

## Resources to refer to

- [Elixir getting started](https://elixir-lang.org/getting-started/basic-operators.html), search on page for `nil`
- [`Kernel.is_nil/1`](https://hexdocs.pm/elixir/Kernel.html#is_nil/1)
- [SO, meaning of `nil`](https://stackoverflow.com/questions/46456704/elixir-what-is-the-meaning-of-nil)

### Hints

- slight hints to guide toward the solution without giving away exact functions/code/implmentation

### After

- nil as a truthy value
- pattern matching on nil
- when to use nil instead of something else

## Representer

- TBD

## Analyzer

- TBD

## Actionable items <a href="#actionable-items" id="actionable-items">🔗</a>

looking for help in a few ways, and I am here to help you through getting this exercise off the ground:

- [ ] implement the [Name Badge](https://github.com/exercism/v3/blob/master/reference/stories/nullability.name-badges.md) narrative to teach nil values
  - Write the exemplar solution (How can we fulfill the premise of the exercise with following the scope of requirements)
  - If we can't fulfill the premise with the current scope, then we need to adjust:
    - the premise
    - the code
    - the requirements
  - let's talk about it! We can help you!
- [ ] Writing the test cases (if TDD is your thing!)
- [ ] Writing the student facing documents
  - I am happy to do this for/with you!

For a first phase review we need:

- [ ] the [design.md document](https://github.com/exercism/v3/blob/master/docs/maintainers/generic-how-to-implement-a-concept-exercise.md#add-metadesignmd-file) (the information in this post)
- [ ] the examplar solution
- [ ] the [introduction document](https://github.com/exercism/v3/blob/master/docs/maintainers/generic-how-to-implement-a-concept-exercise.md#add-docsintroductionmd-file) describing the concepts to be taught

Thanks for considering!

## Need Help?

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

**Comment by angelikatyborska** [comment]
I'll work on this.
